src/main.zig

```diff
@@ -2207,7 +2208,14 @@ pub fn cmdBuild(gpa: *Allocator, arena: *Allocator, args: []const []const u8) !v
                             });
                             fatal("No 'build.zig' file found, in the current directory or any parent directories.", .{});
                         };
-                        continue;
+
+                        //Avoid getting stuck in loop when at root.
+                        if (dirname.len != prevDirname.len) {
+                            prevDirname = dirname;
+                            continue;
+                        }
+
+                        fatal("No 'build.zig' file found, in the current directory or any parent directories.", .{});
```

Comment:
Can you restructure this code a bit? Such that there is no need for the duplicate fatal message and that we also do the `std.log.info` for this path.

Reply:
Made the suggested changes.